cachi2/core/package_managers/rpm/main.py

```diff
@@ -88,7 +88,9 @@ def _resolve_rpm_project(source_dir: RootedPath, output_dir: RootedPath) -> list
         package_dir = output_dir.join_within_root(DEFAULT_PACKAGE_DIR)
         metadata = _download(redhat_rpms_lock, package_dir.path)
         _verify_downloaded(metadata)
-        return _generate_sbom_components(metadata)
+
+        lockfile_relative_path = (source_dir.subpath_from_root / DEFAULT_LOCKFILE_NAME).name
```

Comment:
Even thought the variable is called `source_dir`, it actually contains the package path. So, it will correctly reflect the relative path of the lockfile. I will fix the variable name to avoid further confusion.

Reply:
Right, I wasn't clear in my comment at all, in fact reading it back I got tangled up in my thought process.
With the proposed change you'll always ever end up reporting just the name of the lockfile which is no different to using `DEFAULT_LOCKFILE_NAME` on its own is what I wanted to say I guess, but didn't, sorry, let me illustrate why:

Since we support multiple packages (even for RPM) to be specified on the input:
```
$ pwd
/var/tmp
$ ls foo -1 -R
foo:
rpm_bom_test
rpm_bom_test2

foo/rpm_bom_test:
rpms.lock.yaml

foo/rpm_bom_test2:
rpms.lock.yaml

$ cat /foo/rpm_bom_test/rpms.lock.yaml
lockfileVersion: 1
lockfileVendor: redhat
arches:
  - arch: x86_64
    packages:
      - repoid: ubi-9-baseos-rpms
        url: https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/x86_64/baseos/os/Packages/b/binutils-2.35.2-42.el9_3.1.x86_64.rpm
        checksum: sha256:591050e59decce906f3ba0bcee497fa286acb6c188625913b40c533995d01704
      - repoid: ubi-9-baseos-rpms
        url: https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/x86_64/baseos/os/Packages/b/binutils-gold-2.35.2-42.el9_3.1.x86_64.rpm

$ cat /foo/rpm_bom_test2/rpms.lock.yaml
lockfileVersion: 1
lockfileVendor: redhat
arches:
  - arch: x86_64
    packages:
      - repoid: ubi-9-appstream-rpms
        url: https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/x86_64/appstream/os/Packages/l/libmpc-1.2.1-4.el9.x86_64.rpm
      - repoid: ubi-9-appstream-rpms
        url: https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/x86_64/appstream/os/Packages/l/libxcrypt-devel-4.4.18-3.el9.x86_64.rpm


$ cachi2 fetch-deps --dev-package-managers '[{"type":"rpm", "path":"./foo/rpm_bom_test"},{"type":"rpm", "path":"./foo/rpm_bom_test2"}]'

$ jq . cachi2-output/bom.json
...  
  {
      "name": "binutils-gold",
      "purl": "pkg:rpm/red%20hat%2C%20inc./binutils-gold@2.35.2-42.el9_3.1?arch=x86_64&download_url=https%3A//cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/x86_64/baseos/os/Packages/b/binutils-gold-2.35.2-42.el9_3.1.x86_64.rpm",
      "version": "2.35.2",
      "properties": [
        {
          "name": "cachi2:found_by",
          "value": "cachi2"
        },
        {
          "name": "cachi2:missing_hash:in_file",
          "value": "rpms.lock.yaml"
        }
      ],
      "type": "library"
    },
...
    {
      "name": "libmpc",
      "purl": "pkg:rpm/red%20hat%2C%20inc./libmpc@1.2.1-4.el9?arch=x86_64&download_url=https%3A//cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/x86_64/appstream/os/Packages/l/libmpc-1.2.1-4.el9.x86_64.rpm",
      "version": "1.2.1",
      "properties": [
        {
          "name": "cachi2:found_by",
          "value": "cachi2"
        },
        {
          "name": "cachi2:missing_hash:in_file",
          "value": "rpms.lock.yaml"
        }
      ],
      "type": "library"
    },
...
``` 

So we had 2 packages on the input, each having their own lockfile (yes, very artificial example) with most rpm packages not specifying hashes, so my question is, how do I know which lockfile is missing the hash originating at which input package (hint: in the example above, each of the RPMs comes from a different lockfile) ? 

It's because `.name` will only give you the filename, whereas what I think you wanted was something simple like this instead:
`str(source_dir.subpath_from_root / DEFAULT_LOCKFILE_NAME)`

Reply:
>  I will fix the variable name to avoid further confusion.

Specifically this `source_dir` -> `package_dir` change was completely unnecessary in context of https://github.com/containerbuildsystem/cachi2/pull/525#discussion_r1577502913. FWIW both terms are equally opaque to me with the latter actually being more confusing because package in that context means something completely different - it's not a single RPM package rather a project "subpackage" which isn't really handled in projects differently than main packages, this is usually tackled at the SPEC file level, so I think the original variable name was better, but anyway, I won't NACK the second patch if you think its clearer now.

Reply:
Oh, so it was a bug. You're absolutely right, I fixed it.

I also dropped the renaming commit. And tbh, I never really liked calling of package for what currently is a `*PackageInput` since it can bring confusion in the context of some package managers.